CMakeLists.txt

```diff
@@ -472,8 +472,11 @@ endif()
 option(STATIC "Link libraries statically" ${DEFAULT_STATIC})
 
 # This is a CMake built-in switch that concerns internal libraries
-if (NOT DEFINED BUILD_SHARED_LIBS AND NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
-    set(BUILD_SHARED_LIBS ON)
+set(BUILD_SHARED_LIBS_HELP_TEXT "Build internal libraries as shared")
+if (NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+    option(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS_HELP_TEXT} ON)  # ON for debug, otherwise the build consumes way too much disk space.
```

Comment:
Correct me if I'm wrong but it's not on for static debug, I would edit the comment otherwise the code and the comment don't match.

Reply:
You are right, as long as the external libraries are concerned.

Something like this then?

`# ON for debug & shared external libraries, otherwise the build consumes way too much disk space.`

Reply:
Did you measure the space reduction for debug build with `BUILD_SHARED_LIBS` on and off?

Reply:
Why not just remove all the comments apart from `# This is closest to the final product` ? Unless you tested the storage difference then you can add `# Otherwise the build consumes way too much disk space`

Don't write anything about `ON for debug & shared external libraries` and `OFF by default`, everyone can read the if anyway...

Reply:
In the end you could just add a single line but that's just my preference.

```
option(BUILD_SHARED_LIBS "Build internal libraries as shared" NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug"} 
```

Reply:
> In the end you could just add a single line but that's just my preference.
> 
> ```
> option(BUILD_SHARED_LIBS "Build internal libraries as shared" NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug"} 
> ```

No prob. Will do it so.

Reply:
> Did you measure the space reduction for debug build with `BUILD_SHARED_LIBS` on and off?

I did some time ago, but will do it again to see fresher results.

Reply:
> In the end you could just add a single line but that's just my preference.
> 
> ```
> option(BUILD_SHARED_LIBS "Build internal libraries as shared" NOT STATIC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug"} 
> ```

According to [this post from 2020](https://stackoverflow.com/questions/62487808/can-i-set-a-cmake-variable-to-the-result-of-a-boolean-expression), one-liners aren't possible in CMake.

Reply:
> Did you measure the space reduction for debug build with `BUILD_SHARED_LIBS` on and off?

Measurements are now placed in the description. almost 40% reduction totally.